libcudacxx/include/cuda/std/__iterator/iterator_traits.h

```diff
@@ -170,24 +170,23 @@ struct __iter_traits_cache
 template <class _Iter>
 using _ITER_TRAITS = typename __iter_traits_cache<_Iter>::type;
 
-#if defined(_GLIBCXX_DEBUG)
+#if _CCCL_STD_VER >= 2014
+#  if defined(_GLIBCXX_DEBUG)
 _CCCL_TEMPLATE(class _Iter, class _Ty, class _Range)
 _CCCL_REQUIRES(_IsSame<_Iter, ::__gnu_debug::_Safe_iterator<_Ty*, _Range>>::value)
 _LIBCUDACXX_HIDE_FROM_ABI auto
   __iter_concept_fn(::__gnu_debug::_Safe_iterator<_Ty*, _Range>, __priority_tag<3>) -> contiguous_iterator_tag;
-#endif
-#if defined(__GLIBCXX__)
+#  endif
+#  if defined(__GLIBCXX__)
 _CCCL_TEMPLATE(class _Iter, class _Ty, class _Range)
 _CCCL_REQUIRES(_IsSame<_Iter, ::__gnu_cxx::__normal_iterator<_Ty*, _Range>>::value)
 _LIBCUDACXX_HIDE_FROM_ABI auto
   __iter_concept_fn(::__gnu_cxx::__normal_iterator<_Ty*, _Range>, __priority_tag<3>) -> contiguous_iterator_tag;
-#endif
-#if defined(_LIBCPP_VERSION)
+#  elif defined(_LIBCPP_VERSION)
```

Comment:
I am sorry, but my comment was erroneous.

We always pull in libstdc++ through some nvcc machinery, so we need to make this a separate branch